### PR TITLE
Use TensorBuilder in TensorCollector

### PR DESCRIPTION
**Pull request by danieldk**
In order to do so, extend TensorBuilder to also support collection of
labels.

### CODE REVIEW

File: sticker/src/tensorflow/tensor.rs [hunk 1]
```diff
@@ -42,4 +74,34 @@ impl TensorBuilder {
...
+        &self.labels
+    }
+}
+
+impl TensorBuilder<LabelTensor> {
+    pub fn add_with_labels(&mut self, input: &[f32], labels: &[i32]) {
+        self.add_input(input);
+
+        let max_seq_len = self.inputs.dims()[1] as usize;
+        let token_embed_size = self.inputs.dims()[2] as usize;
```
Comment (sebpuetz) [inline sticker/src/tensorflow/tensor.rs]: That's not only the dimensionality of the token embeddings but might also include other embeddings, right?

File: sticker/src/tensorflow/tensor.rs [hunk 2]
```diff
@@ -31,8 +61,10 @@ impl TensorBuilder {
         let token_seq =
             &mut self.inputs[token_offset..token_offset + (token_embed_size * timesteps)];
         token_seq.copy_from_slice(&input[..token_embed_size * timesteps]);
```
Comment (sebpuetz) [inline sticker/src/tensorflow/tensor.rs]: `token_embed_size` is somewhat ambiguous.

**Review by sebpuetz — APPROVED** [review]
Looks good to me, although there's some ambiguity with `token_embed_size`. Could be read as the size of the token embedding or the size of the embeddings associated with a token.

**Comment by danieldk** [comment]
> Looks good to me, although there's some ambiguity with `token_embed_size`. Could be read as the size of the token embedding or the size of the embeddings associated with a token.

"There are only two hard things in Computer Science: cache invalidation and naming things" -- Phil Karlton

I have changed the name to `token_repr_size`. 'Token representation' is hopefully a bit less ambiguous than 'token embedding'.

**Comment by danieldk** [comment]
Thanks for the review! Will merge once Travis-CI is happy.

**Pull request merged**